### ISSUE TITLE
Test discovery

### ISSUE THREAD

**Issue by DorianCzichotzki** [darrenburns/ward#129]
It would be nice to just list all tests ward found without directly running them.
Something like a `--discover` flag for the ward command.

**Comment by darrenburns** [comment]
I could see this being useful, and I'd accept a PR for it :)

Some other suggestions for names: `--collect-only`, `--print-tests`, `--print-only`, `--dry-run`.

**Comment by DorianCzichotzki** [comment]
Would it be sufficient to add a DRYRUN/PRINT (a better name is welcome) TestOutcome and change the Test.run method @darrenburns?

I am currently working with something like this:
```python
# TODO:onlyanegg: I don't love this. We're setting up the
# fixture within the testing module, but cleaning it up in the
# suite module.
resolved_args = self.resolve_args(
    cache, iteration=self.param_meta.instance_index
)
self.format_description(resolved_args)
if dry_run:
    with closing(self.sout), closing(self.serr):
        result = TestResult(self, TestOutcome.DRYRUN)
    return result
self.fn(**resolved_args)
```

Or alternatively just return PASS and SKIP for all tests?

**Comment by darrenburns** [comment]
I like the idea of returning an outcome of `DRYRUN` - I think that'd be sufficient, yes.

**Comment by darrenburns** [comment]
Closing this as `--dry-run` is now available in 0.36.0b0
